scripts/__input_class_gamepad_motion/__input_class_gamepad_motion.gml

```diff
@@ -0,0 +1,138 @@
+                __gyro_y = -degtorad(switch_controller_angular_velocity(__index, switch_controller_axis_z, _sensor)/5) * 570.6;
+                __gyro_z = _sign_z * degtorad(switch_controller_angular_velocity(__index, _axis_z, _sensor)/5) * 570.6;
+            break;            
+
+            case (os_ps4):
+            case (os_ps5):
+                __accel_x =  gamepad_axis_value(__index, gp_axis_acceleration_x);
+                __accel_y = -gamepad_axis_value(__index, gp_axis_acceleration_y);
+                __accel_z = -gamepad_axis_value(__index, gp_axis_acceleration_z);
+                
+                __gyro_x =  gamepad_axis_value(__index, gp_axis_angular_velocity_x) / pi;
+                __gyro_y = -gamepad_axis_value(__index, gp_axis_angular_velocity_y) / pi;
+                __gyro_z = -gamepad_axis_value(__index, gp_axis_angular_velocity_z) / pi;
+            break;
+
+            case (os_windows):
+            case (os_linux):
+                var _steam_handle = global.__input_gamepads[__index].__steam_handle;
+                if (_steam_handle != undefined)
+                {                    
+                    var _steam_data = steam_input_get_motion_data(_steam_handle);
+                    if (!is_struct(_steam_data)) 
+                    {
+                        return undefined;
+                        break;
+                    }
+                    
+                    __accel_x =  _steam_data.pos_accel_x / 16384;
+                    __accel_y = -_steam_data.pos_accel_y / 16384;
+                    __accel_z = -_steam_data.pos_accel_z / 16384;
+                    
+                    var _gyro_scale = 5898.24;
+                    if ((global.__input_gamepads[__index].raw_type == "SteamController")
+                    ||  (global.__input_gamepads[__index].raw_type == "CommunityDeck"))
+                    {
+                        _gyro_scale /= 2;
+                    }
+                    
+                    __gyro_x =  _steam_data.rot_vel_x / _gyro_scale;
+                    __gyro_y = -_steam_data.rot_vel_z / _gyro_scale;
+                    __gyro_z = -_steam_data.rot_vel_y / _gyro_scale;
+                }                
+            break;
+
+            default: return undefined; break;
+        }
+        
+        return __get_data();
+    };
+    
+    static __clear = function()
+    {        
+        __accel_x =  0.0;
+        __accel_y = -1.0;
+        __accel_z =  0.0;
+
+        __gyro_x = 0.0;
+        __gyro_y = 0.0;
+        __gyro_z = 0.0;
+    };
+
+    static __get_data = function()
+    {
+        return {
+            acceleration_x: __accel_x,
+            acceleration_y: __accel_y,
+            acceleration_z: __accel_z,
+
+            angular_velocity_x: __gyro_x,
+            angular_velocity_y: __gyro_y,
+            angular_velocity_z: __gyro_z,
+        }
```

Comment:
Consider making this struct static otherwise we're spinning up a struct for every call of `__get_data()`